### DISCUSSION TITLE
remove 'invalid date' when no date is entered?

### DISCUSSION THREAD

**Discussion by zeltak** [chhoumann/quickadd#567] [General]
Hi I use quick add a lot to enter tasks into obsidian with the following template:

```
#task {{VALUE:task name}} 📅 {{VDATE:due date,YYYY-MM-DD}} {{VALUE:tags}}
```

this first asks for date (wonder why it dosent do that after task name but thats another issue ;-) )

sometime I dont want a date and I leave the input for due date empty. but this instead of giving an empty date generates something like this:

```
- [ ] #task create amazon trackers for items i need 📅 Invalid date #home #shop
```

any clue how this can be resolved?

best

Z

**Comment by FeralFlora** [comment]
I think this relates to this issue: https://github.com/chhoumann/quickadd/issues/448

**Comment by chhoumann** [comment]
Coming back to this one very late - sorry about that.

This is built in now: the `|optional` flag shipped in QuickAdd 2.13.0 (closing the #448 that @FeralFlora linked). Your template becomes:

```
#task {{VALUE:task name}} {{VDATE:due date,[📅 ]YYYY-MM-DD|optional}} {{VALUE:tags}}
```

Leave the date prompt empty (or hit Skip) and the token resolves to nothing instead of `Invalid date`. Note the `[📅 ]` moved *inside* the date format: square brackets mark literal text, so the emoji renders with a date and vanishes with a skipped one - no stray 📅 on undated tasks.

On your aside about the date being asked first: if you turn on **One-page input for choices (Beta)** in QuickAdd's settings, you get a single form with all the fields at once instead of a sequence of prompts, so the ordering stops mattering.

Docs: https://quickadd.obsidian.guide/docs/FormatSyntax#optional-fields

Closing as resolved.
